Doc/library/asyncio-task.rst

```diff
@@ -414,6 +414,73 @@ reported by :meth:`asyncio.Task.cancelling`.
    Improved handling of simultaneous internal and external cancellations
    and correct preservation of cancellation counts.
 
+Group Cancellation
+------------------
+
+Cancelling an entire task group is not natively supported by the standard
+library but can be achieved by adding a task to the group that raises an
+exception and ignore that exception:
+
+.. code-block:: python
+
+   import contextlib
+   from asyncio import TaskGroup
+
+   class CancelTaskGroup(Exception):
+      """Exception raised to cancel a task group."""
+
+   @contextlib.asynccontextmanager
+   async def CancellableTaskGroup():
+      """A cancellable task group."""
+      try:
+          async with TaskGroup() as group:
+              yield group
+      except* CancelTaskGroup:
+          print('task group was cancelled')
+
+   async def cancel_group(cancellable_task_group):
+      """Create a task that will cancel the group it belongs to."""
+      async def factory():
+          raise CancelTaskGroup(cancellable_task_group)
+      await cancellable_task_group.create_task(factory())
+
+The above helper functions can be used as follows:
```

Comment:
This whole block looks way too long. I think all we need is the suggestion I made above.